dao/mongo.go

```diff
@@ -300,3 +306,25 @@ func (m *MongoService) GetPaymentsWithRefundPendingStatus() ([]models.PaymentRes
 
 	return payments, nil
 }
+
+func (m *MongoService) PatchPaymentsWithRefundPendingStatus(id string, paymentUpdate *models.PaymentResourceDB) error {
+	collection := m.db.Collection(m.CollectionName)
+
+	refundsFilter := options.ArrayFilters{Filters: bson.A{bson.M{"x.refund_id": paymentUpdate.Refunds[0].RefundId}}}
+	upsert := true
+
+	opts := options.UpdateOptions{
+		ArrayFilters: &refundsFilter,
+		Upsert:       &upsert,
```

Comment:
Why is Upsert being set to true?

Reply:
This enabled both update and insert to happen simultaneously without issue.